File: assignments_nilesh_sir/java_assignments/Assignment_Bank/Banking/HomeLoan.java
package Banking;

public class HomeLoan extends Loan implements Discountable {
    private double limit = 250000.0;

    public double getRate() {
        if (this.getPrinciple() >= limit) {
            return this.getPrinciple() >= 2000000.0 ? 0.11 : 0.12;
        } else {
            return this.getPrinciple() >= 2000000.0 ? 0.10 : 0.11;
        }
    }

    public double getDiscount() {
        return this.getEMI() > 1500.0 ? this.getEMI() * 0.05 : 0.0;
    }
}

File: Dotnet/04-05-24/Assignment_3/BankLib/HomeLoan.cs
namespace BankLib
{

    sealed class HomeLoan : Loan, IDiscountable
    {

        double limit = 0;
        public HomeLoan()
        {
            limit = 2500000;
        }
        public override double GetRate()
        {
            double rate = Principle <= 2000000 ? 0.10 : 0.11;
            return Principle > limit ? rate += 0.01 : rate;
        }

        public double GetDiscount()
        {
            return GetEmi() * 0.05;
        }
    }
}